webapp-django/crashstats/topcrashers/views.py

```diff
@@ -180,21 +182,25 @@ def get_topcrashers_results(**kwargs):
 def topcrashers(request, days=None, possible_days=None, default_context=None):
     context = default_context or {}
 
-    product = request.GET.get('product')
-    versions = request.GET.getlist('version')
-    crash_type = request.GET.get('process_type')
-    os_name = request.GET.get('platform')
-    result_count = request.GET.get('_facets_size')
-    tcbs_mode = request.GET.get('_tcbs_mode')
-    range_type = request.GET.get('_range_type')
+    form = TopCrashersForm(request.GET)
+    if not form.is_valid():
+        return http.HttpResponseBadRequest(unicode(form.errors))
+
+    product = form.cleaned_data['product']
+    versions = form.cleaned_data['version']
+    crash_type = form.cleaned_data['process_type']
+    os_name = form.cleaned_data['platform']
+    result_count = form.cleaned_data['_facets_size']
+    tcbs_mode = form.cleaned_data['_tcbs_mode']
+    range_type = form.cleaned_data['_range_type']
 
     range_type = 'build' if range_type == 'build' else 'report'
 
     if not tcbs_mode or tcbs_mode not in ('realtime', 'byday'):
         tcbs_mode = 'realtime'
 
     if product not in context['active_versions']:
-        raise http.Http404('Unrecognized product')
+        return http.HttpResponseBadRequest('Unrecognized product')
```

Comment:
This was slightly off-topic but I saw it and it was easy to fix. 
We shouldn't say that the page doesn't exist when it did but with the parameters being wrong.